__tests__/app.test.js

```diff
@@ -52,6 +53,84 @@ describe('topics', () =>
 
 describe('articles', () =>
 {
+    describe.only('GET /api/articles', () =>
+    {
+        test('200: responds with array of articles', () =>
+        {
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then(({body}) =>
+            {
+                const {articles} = body;
+                expect(articles).toBeInstanceOf(Array);
+                expect(articles.length).toBe(12);
+                articles.forEach(article =>
+                {
+                    expect(article).toEqual(expect.objectContaining
+                    ({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number)
+                    }));
+                });
+            });
+        });
+        test('200: sorts array by date in DESC order by default', () =>
+        {
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then(({body}) =>
+            {
+                const {articles} = body;
+                expect(articles).toBeSortedBy('created_at', {descending: true});
+            });
+        });
+        test('200: sorts array by specified column', () =>
+        {
+            return request(app)
+            .get('/api/articles?sort_by=author')
+            .expect(200)
+            .then(({body}) =>
+            {
+                const {articles} = body;
+                expect(articles).toBeSortedBy("author", {descending:true});
+            });
+        });
+        test('200: orders array by specified parameter', () =>
+        {
+            return request(app)
+            .get('/api/articles?order=ASC')
+            .expect(200)
+            .then(({body}) =>
+            {
+                const {articles} = body;
+                expect(articles).toBeSortedBy("created_at", {descending:false});
+            });
+        });
+
+        test('200: filters by specified topic query', () =>
+        {
+            return request(app)
+            .get('/api/articles?topic=mitch')
+            .expect(200)
+            .then(({body}) =>
+            {
+                const {articles} = body;
+                expect(articles.length).toBe(11);
+                articles.forEach(article =>
+                {
+                    expect(article.topic).toBe("mitch");
+                });
+            })
+        });
+    });
+
     describe('GET /api/articles/:article_id', () =>
     {
         test('200: responds with object containing expected values', () =>
```

Comment:
Nice work on these endpoints. I'd consider adding a test here for an invalid sort_by here.